src/workerd/api/actor-state.c++

```diff
@@ -919,6 +919,37 @@ kj::Maybe<kj::Date> DurableObjectState::getWebSocketAutoResponseTimestamp(jsg::R
   return ws->getAutoResponseTimestamp();
 }
 
+void DurableObjectState::setHibernatableWebSocketEventTimeout(kj::Maybe<int> timeoutMs) {
+  auto& a = KJ_REQUIRE_NONNULL(IoContext::current().getActor());
+
+  KJ_IF_SOME(timeout, timeoutMs) {
+    if (timeout > 0) {
+      KJ_IF_SOME(hibernationManager, a.getHibernationManager()) {
+        hibernationManager.setEventTimeout(timeout);
+      } else {
+        a.setHibernationManager(kj::refcounted<HibernationManagerImpl>(
+            a.getLoopback(), KJ_REQUIRE_NONNULL(a.getHibernationEventType(), timeout)));
+      }
+      return;
+    }
+  }
+
+  // Setting a timeout <=0ms or an empty value will unset any currently set event timeout.
```

Comment:
I left comments elsewhere but I don't think negative values should be passed around at all, and I definitely don't think setting a negative value should clear any existing timeout. Allowing only positive values (or `kj::none` to unset) should make this easier to follow.

At the very least if we stick with `int`s then I think negative values should throw.